src/vs/base/browser/ui/tree/indexTreeModel.ts

```diff
@@ -219,7 +219,7 @@ export class IndexTreeModel<T extends Exclude<any, undefined>, TFilterData = voi
 
 		const result = this._setListNodeCollapsed(node, listIndex, revealed, collapsed!, recursive || false);
 
-		if (this.autoExpandSingleChildren && !collapsed! && !recursive) {
+		if (node !== this.root && this.autoExpandSingleChildren && !collapsed! && !recursive) {
```

Comment:
This fixes the bug from the second gif from https://github.com/microsoft/vscode/issues/78153#issuecomment-522271070

Reply:
Can you please put this in a seperate PR. I like this simple change and it is in the tree land so I would love if we merge it in unrelated to this.

Reply:
Ok

Reply:
Opened #79540 and it's corresponding PR #79542